File: mycollab-api/src/main/java/com/mycollab/rest/server/resource/CheckVersionController.java
package com.mycollab.rest.server.resource;

import com.mycollab.core.Version;
import com.mycollab.ondemand.module.billing.dao.ProEditionInfoMapper;
import com.mycollab.ondemand.module.support.service.EditionInfoResolver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.Properties;

/**
 * @author MyCollab Ltd
 * @since 5.1.2
 */
@RestController
public class CheckVersionController {

    @Autowired
    private EditionInfoResolver editionInfoResolver;

    @RequestMapping(value = "/checkupdate", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public Properties getLatestVersion(@RequestParam("version") String version) {
        Properties props = new Properties();

        String liveVersion = editionInfoResolver.getEditionInfo().getVersion();
        props.put("version", liveVersion);
        props.put("downloadLink", "https://www.mycollab.com/ce-registration/");
        props.put("releaseNotes", String.format("https://community.mycollab.com/releases/release-notes-for-mycollab-%s/",
                Version.getVersion().replace('.', '-')));

        if (version != null && Version.isEditionNewer(liveVersion, version) &&
                Version.isEditionNewer(version, "5.3.4")) {
            props.put("autoDownload", editionInfoResolver.getEditionInfo().getCommunityUpgradeLink());
        }

        return props;
    }

    @Autowired
    private ProEditionInfoMapper proEditionInfoMapper;

    @RequestMapping(value = "/checkpremiumupdate", method = RequestMethod.GET)
    public Properties getLatestPremiumUpdate(@RequestParam("version") String version, @RequestParam("customerId") String customerId) {
        Properties props = new Properties();
        String liveVersion = editionInfoResolver.getEditionInfo().getVersion();
        props.put("version", liveVersion);
        props.put("downloadLink", "https://www.mycollab.com/ee-registration/");
        props.put("releaseNotes", String.format("https://community.mycollab.com/releases/release-notes-for-mycollab-%s/",
                Version.getVersion().replace('.', '-')));
        props.put("autoDownload", editionInfoResolver.getEditionInfo().getPremiumUpgradeLink());
        return props;
    }
}

File: mycollab-api/src/main/java/com/mycollab/rest/server/resource/CheckVersionController.kt
package com.mycollab.rest.server.resource

import com.mycollab.core.Version
import com.mycollab.ondemand.configuration.EditionInfo
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.util.*

/**
 * @author MyCollab Ltd
 * @since 5.1.2
 */
@Api(value = "Versions", tags = ["Support"])
@RestController
class CheckVersionController(private val editionInfo: EditionInfo) {

    @ApiOperation(value = "Check version whether it is the latest version. If it is not, return the latest version information", response = String::class)
    @RequestMapping(value = ["/checkupdate"], method = [(RequestMethod.GET)], produces = [(MediaType.APPLICATION_JSON_VALUE)])
    fun getLatestVersion(@RequestParam("version", required = false) version: String?): Properties {
        val props = Properties()

        val liveVersion = editionInfo.version
        props["version"] = liveVersion
        props["downloadLink"] = "https://www.mycollab.com/ce-registration/"
        props["releaseNotes"] = "https://community.mycollab.com/releases/release-notes-for-mycollab-${Version.getVersion().replace('.', '-')}/"

        if (version != null && Version.isEditionNewer(liveVersion, version) &&
                Version.isEditionNewer(version, "5.3.4")) {
            props["autoDownload"] = editionInfo.communityUpgradeLink
        }

        return props
    }

    @RequestMapping(value = ["/checkpremiumupdate"], method = [(RequestMethod.GET)])
    fun getLatestPremiumUpdate(@RequestParam("version") version: String, @RequestParam("customerId") customerId: String): Properties {
        val props = Properties()
        val liveVersion = editionInfo.version
        props["version"] = liveVersion
        props["downloadLink"] = "https://www.mycollab.com/ee-registration/"
        props["releaseNotes"] = "https://community.mycollab.com/releases/release-notes-for-mycollab-${Version.getVersion().replace('.', '-')}/"
        props["autoDownload"] = editionInfo.premiumUpgradeLink
        return props
    }
}
